source/rally_and_tempest.rst

```diff
@@ -0,0 +1,97 @@
+.. include:: vars.rst
+
+==========================================
+Verifying the Cloud with Rally and Tempest
+==========================================
+
+`Rally <https://docs.openstack.org/rally/latest/>`_ is a test framework,
+and `Tempest <https://docs.openstack.org/tempest/latest/>`_ is the
+OpenStack API test suite.  In this guide, Rally is used to run Tempest tests.
+
+Requirements
+------------
+
+OpenStack tests are run from a host with access to the OpenStack APIs
+and external network for instances.
+
+The following software environment is needed:
+
+* OpenStack admin credentials, eg public-openrc.sh from kayobe-config/etc/kolla
+* A virtualenv setup with python-openstackclient installed
+
+Setup Rally for a new user
+--------------------------
+
+A good directory hierarchy would be `~/rally/shakespeare/tempest-recipes`.
+This is assumed in this guide.
+
+Install Rally into the virtualenv:
+
+.. code-block:: shell
+
+   pip3 install git+https://github.com/openstack/rally-openstack.git \
+        --constraint https://raw.githubusercontent.com/openstack/rally-openstack/master/upper-constraints.txt
+
+Create the Rally test database and configuration file:
+
+.. code-block:: shell
+
+   mkdir -p ~/.rally ~/rally/data
+   echo "[database]" | tee ~/.rally/rally.conf
+   echo "connection=sqlite:////home/${USER}/rally/data/rally.db" | tee -a ~/.rally/rally.conf
```

Comment:
This makes assumptions about location of home directories. Use `${HOME}` instead.